### ISSUE TITLE
Mirrored strategy results in [Error: Invalid loss, terminating training]

### ISSUE THREAD

**Issue by jasony93** [HamadYA/GhostFaceNets#60]
First of all, I appreciate for this great work.
Since the dataset is too big, I am trying to use 4 gpus to shorten the training time.
When I train with single gpu, everything works fine, but when I train with multiple gpus, the training stops and prints "Error: Invalid loss, terminating training".
What I tried was to scale the loss and learning late so that the loss does not go to infinity, but this does not seem to work.
Eventually, I found somewhat reasonable stackoverflow post below.
https://stackoverflow.com/questions/54283937/training-on-multiple-gpus-causes-nan-validation-errors-in-keras
Basically if the number of data is not divisible by batch size and number of gpus, then empty batches will be passed to the loss function.
I tried to apply this code, and it seems to work for a while, but it does not seem to train properly. (the loss does not decrease)
Do you have any idea?

Thanks in advance.

**Comment by HamadYA** [comment]
> I tried to apply this code, and it seems to work for a while, but it does not seem to train properly. (the loss does not decrease)

Hello,
Thanks for your comment.

Actually I have never trained this model with multi-gpu as I have no access to any.
